File: DS/SingleLinkList.py
class LinkList:

	def __init__(self):
		self.head = None
		
	def addToLast(self, nn):
		if(self.head == None):
			self.head = nn
		else:
			temp = self.head
			while(temp.next != None):
				temp = temp.next
			temp.next = nn
				
	def display(self):
		temp = self.head
		while(temp != None):
			print(temp.data)
			temp = temp.next

class Node:

	def __init__(self, data):
		self.data = data;
		self.next = None;


def main():
	ll = LinkList()
	nn = Node(10)
	ll.addToLast(nn)
	ll.addToLast(Node(20))
	ll.display()

if __name__ == '__main__':
	main()

File: DS/SingleLinkList.java
class LinkList{
	Node head;
	public void addToLast(Node nn){
			if(head == null){
				head = nn;
			}
			else{
				Node temp = head;
				while(temp.next != null){
					temp = temp.next;
				}
				temp.next = nn;
			}
		}
	public void display(){
		Node temp = head;
		while(temp != null){
			System.out.println(temp.data);
			temp = temp.next;
		}
	}
}

class Node{
	int data;
	Node next;
	Node(int data){
		this.data = data;
	}
}

class SingleLinkList{
	public static void main(String args[]){
	LinkList ll = new LinkList();
	Node nn = new Node(10);

	ll.addToLast(nn);
	ll.addToLast(new Node(20));
	ll.display();
	
	}
}
